lib/std/child_process.zig

```diff
@@ -257,6 +259,55 @@ pub const ChildProcess = struct {
         }
     }
 
+    fn collectOutputWindows(child: *ChildProcess, outs: [2]*std.ArrayList(u8), max_output_bytes: usize) !void {
+        const bump_amt = 512;
+        const handles = [_]windows.HANDLE{ child.stdout.?.handle, child.stderr.?.handle };
+
+        for ([_]u1{ 0, 1 }) |i| {
+            try outs[i].ensureCapacity(bump_amt);
+            const buf = outs[i].unusedCapacitySlice();
+            child.overlapped[i] = mem.zeroes(windows.OVERLAPPED);
+            _ = windows.kernel32.ReadFile(handles[i], buf.ptr, @intCast(u32, buf.len), null, &child.overlapped[i]);
+        }
+
+        var wait_objects = handles;
+        var wait_object_count: u2 = wait_objects.len;
+
+        while (true) {
+            const status = windows.kernel32.WaitForMultipleObjects(wait_object_count, &wait_objects, 0, windows.INFINITE);
+            if (status == windows.WAIT_FAILED) {
+                switch (windows.kernel32.GetLastError()) {
+                    else => |err| return windows.unexpectedError(err),
+                }
+            }
+            if (status < windows.WAIT_OBJECT_0 or status > windows.WAIT_OBJECT_0 + wait_object_count - 1)
+                unreachable;
+
+            const wait_idx = status - windows.WAIT_OBJECT_0;
+            const i = @as(u1, if (wait_objects[wait_idx] == handles[0]) 0 else 1);
+            var read_bytes: u32 = undefined;
+            const overlapped_result = windows.kernel32.GetOverlappedResult(handles[i], &child.overlapped[i], &read_bytes, 0);
+            if (overlapped_result == 0) switch (windows.kernel32.GetLastError()) {
+                .BROKEN_PIPE => {
+                    wait_object_count -= 1;
+                    if (wait_object_count == 0)
+                        break;
+                    if (wait_idx == 0)
+                        wait_objects[0] = wait_objects[1];
+                    continue;
+                },
+                else => |err| return windows.unexpectedError(err),
+            };
+
+            outs[i].items.len += read_bytes;
+            const new_capacity = std.math.min(outs[i].items.len + bump_amt, max_output_bytes);
+            try outs[i].ensureCapacity(new_capacity);
+            const buf = outs[i].unusedCapacitySlice();
+            if (buf.len == 0) return if (i == 0) error.StdoutStreamTooLong else error.StderrStreamTooLong;
```

Comment:
Style nit, let's avoid overlong lines.
```zig
if (buf.len == 0) {
    return if (i == 0) error.StdoutStreamTooLong else error.StderrStreamTooLong;
}
```

Reply:
I tried breaking this up but `zig fmt` brought it back to one long line

Reply:
Try harder, don't let the machine win.

Reply:
If you want a style change, rather than me wasting my time trying to figure out exactly what you want, why don't you just tell me?

Reply:
> Style nit, let's avoid overlong lines.

Reply:
how long do you want lines to be?  there are quite a few lines longer than this one, so are you asking me to cut all lines in this new code to some limit?